app.js

```diff
@@ -599,24 +618,32 @@ define(function(require) {
 			});
 
 			template.find('.account-settings-update').on('click', function() {
+				// Validate form
 				if (!monster.ui.valid(template.find('#account_settings_form'))) {
 					return;
 				}
 
+				// Collect data
 				var formData = monster.ui.getFormData('account_settings_form'),
-					newData = $.extend(true, {}, data.account, formData);
+					newData = _.merge({}, data.account, formData);
+
+				// Format data
+				if (_.trim(newData.caller_id.asserted.number)) {
```

Comment:
You are right, it will be validated by the `phoneNumber` in case it is not empty but full of empty spaces. I'll change it with an `lodash#isEmpty` check, to see if it is an empty string, because it is an optional field. I want to keep that empty value instead of the `undefined` value that `monster.util.getFormatPhoneNumber().e164Number` will return, in order for it not to be ignored when the form data is merged with the original data.